src/core/parse_feeder.hpp

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include <string>
+#include <nlohmann/json.hpp>
+#include "token.hpp"
+#include "symbol.hpp"
+
+namespace docgen {
+namespace core {
+
+struct ParseFeeder {
+	private:
+		static nlohmann::json parsed_, parsing_;
+		static std::string key_;
+		static bool writing_;
+		static bool just_written_;
+		static bool to_skip_;
+		static std::string *val_()
+		{
+			if (parsing_[key_].is_null()) {
+				parsing_[key_] = "";
+			}
+			return parsing_[key_].get_ptr<std::string *>();
+		}
+		static void short_reset()
+		{
+			parsing_.clear();
+			key_.clear();
+			writing_ = false;
+			just_written_ = false;
+			to_skip_ = false;
+		}
+
+	public:
+		static void go() { writing_ = true; just_written_ = false; }
```

Comment:
Can we rename `go` to `start`, `(second) at` to `is_key_nonempty` (although I prefer to use `is_key_empty` and everywhere you used it just put `!is_key_empty()`, `on` to something else (can't think of a good name for this one yet)

Reply:
eh

Reply:
they're not that intuitive, I had to constantly look at the definition. Especially because you have two different meanings for `at`.

Reply:
? The `at()` thing is actually more of a C++ convention—that is how getters and setters typically look in C++. Overloading the same function s.t. no parameter is the getter. I've seen that a lot actually.

I don't like `start` because it implies a one-time thing to me. `resume` would be more appropriate, but still not perfect since the `ParseFeeder` begins in a stopped state. This is why I like `stop` and `go` the best—they are the most suggestive of a traffic-like situation, which is most fitting for this.

Reply:
The at function typically returns a reference of an element to get. You set by modifying this reference like 

m.at(1) = 89;

Your at function directly sets and is never a getter. Your second at doesn't even do either but checks if key is empty.

Reply:
Lol u right. I was being cute with the names (they were originally temporary when I wasn't sure this was the way to go), but I'll make them more descriptive.

Reply:
Well I still like `go` rather than `start`, but I'll see if I can think of something more descriptive.